### PR TITLE
Reorg analytics instrumentation/roots/visibility to support FIE ampdocs

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

File: src/service.js [hunk 1]
```diff
@@ -323,11 +323,12 @@ export function getTopWindow(win) {
 /**
  * Returns the parent "friendly" iframe if the node belongs to a child window.
  * @param {!Node} node
- * @param {!Window} topWin
+ * @param {!Window=} opt_topWin
  * @return {?HTMLIFrameElement}
  */
-export function getParentWindowFrameElement(node, topWin) {
+export function getParentWindowFrameElement(node, opt_topWin) {
```
Comment (lannka) [inline src/service.js]: seems to me quite random to have this method here. It's only used by amp-analytics.
We can either move it to amp-analytics, or move it to friendly-iframe-embed.js for future reuse.

Comment (dvoytenko) [reply src/service.js]: Hmm. It's used in many places.

Comment (lannka) [reply src/service.js]: You're right. Still be good to move out to friendly-iframe-embed.js. but can be a different PR

File: extensions/amp-analytics/0.1/instrumentation.js [hunk 2]
```diff
@@ -89,24 +89,20 @@ export class InstrumentationService {
    * @return {!./analytics-root.AnalyticsRoot}
    */
   findRoot_(context) {
-    // FIE
-    const frame = getParentWindowFrameElement(context, this.ampdoc.win);
-    if (frame) {
-      const embed = getFriendlyIframeEmbedOptional(frame);
+    // TODO(#22733): cleanup when ampdoc-fie is launched.
```
Comment (lannka) [inline extensions/amp-analytics/0.1/instrumentation.js]: could you be more clear about this TODO. My understanding that before ampdoc-fie launch the ampdoc returned by L93 will still be the same, so we need `embed` to be the holder. After the launch, embed can be removed from this method?

**Comment by lannka** [comment]
Just realized that we don't have integration test for visible trigger in FIE. Pls hold on and I'll manage to get a test in this afternoon.

**Comment by dvoytenko** [comment]
@lannka No prob. LMK.

**Comment by lannka** [comment]
test PR is here: https://github.com/ampproject/amphtml/pull/23438

**Comment by lannka** [comment]
FYI, the test is merged. YOu're ready to go

**Pull request merged**